libraries/Ticker/src/Ticker.h

```diff
@@ -26,82 +26,115 @@
 #define TICKER_H
 
 extern "C" {
-  #include "esp_timer.h"
+#include "esp_timer.h"
 }
+#include <functional>
 
 class Ticker
 {
 public:
-  Ticker();
-  ~Ticker();
-  typedef void (*callback_t)(void);
-  typedef void (*callback_with_arg_t)(void*);
-
-  void attach(float seconds, callback_t callback)
-  {
-    _attach_ms(seconds * 1000, true, reinterpret_cast<callback_with_arg_t>(callback), 0);
-  }
-
-  void attach_ms(uint32_t milliseconds, callback_t callback)
-  {
-    _attach_ms(milliseconds, true, reinterpret_cast<callback_with_arg_t>(callback), 0);
-  }
-
-  template<typename TArg>
-  void attach(float seconds, void (*callback)(TArg), TArg arg)
-  {
-    static_assert(sizeof(TArg) <= sizeof(uint32_t), "attach() callback argument size must be <= 4 bytes");
-    // C-cast serves two purposes:
-    // static_cast for smaller integer types,
-    // reinterpret_cast + const_cast for pointer types
-    uint32_t arg32 = (uint32_t)arg;
-    _attach_ms(seconds * 1000, true, reinterpret_cast<callback_with_arg_t>(callback), arg32);
-  }
-
-  template<typename TArg>
-  void attach_ms(uint32_t milliseconds, void (*callback)(TArg), TArg arg)
-  {
-    static_assert(sizeof(TArg) <= sizeof(uint32_t), "attach_ms() callback argument size must be <= 4 bytes");
-    uint32_t arg32 = (uint32_t)arg;
-    _attach_ms(milliseconds, true, reinterpret_cast<callback_with_arg_t>(callback), arg32);
-  }
-
-  void once(float seconds, callback_t callback)
-  {
-    _attach_ms(seconds * 1000, false, reinterpret_cast<callback_with_arg_t>(callback), 0);
-  }
-
-  void once_ms(uint32_t milliseconds, callback_t callback)
-  {
-    _attach_ms(milliseconds, false, reinterpret_cast<callback_with_arg_t>(callback), 0);	
-  }
-
-  template<typename TArg>
-  void once(float seconds, void (*callback)(TArg), TArg arg)
-  {
-    static_assert(sizeof(TArg) <= sizeof(uint32_t), "attach() callback argument size must be <= 4 bytes");
-    uint32_t arg32 = (uint32_t)(arg);
-    _attach_ms(seconds * 1000, false, reinterpret_cast<callback_with_arg_t>(callback), arg32);
-  }
-
-  template<typename TArg>
-  void once_ms(uint32_t milliseconds, void (*callback)(TArg), TArg arg)
-  {
-    static_assert(sizeof(TArg) <= sizeof(uint32_t), "attach_ms() callback argument size must be <= 4 bytes");
-    uint32_t arg32 = (uint32_t)(arg);
-    _attach_ms(milliseconds, false, reinterpret_cast<callback_with_arg_t>(callback), arg32);
-  }
-
-  void detach();
-  bool active();
-
-protected:	
-  void _attach_ms(uint32_t milliseconds, bool repeat, callback_with_arg_t callback, uint32_t arg);
-
+	Ticker();
```

Comment:
When you change the style of the header like this, it get's hard to track what actual changes you have made. Was the change to spacing necessary?